botocore/credentials.py

```diff
@@ -178,6 +181,51 @@ def __call__(self):
     return _Refresher(actual_refresh)
 
 
+class JSONFileCache(object):
+    """JSON file cache.
+    This provides a dict like interface that stores JSON serializable
+    objects.
+    The objects are serialized to JSON and stored in a file.  These
+    values can be retrieved at a later time.
+    """
+
+    CACHE_DIR = os.path.expanduser(os.path.join('~', '.aws', 'cli', 'cache'))
```

Comment:
I think we should consider using a different directory than what the CLI uses. I have not thought through whether there will be edge cases to this, but it may be surprising if you are running a script to learn that your temp credentials were being cached under a `cli` directory. I wonder if it makes sense to call it `boto` or `python` instead?

Reply:
sounds reasonable

Reply:
done